app/assets/javascripts/labels_index.coffee

```diff
@@ -0,0 +1,179 @@
+        more_info: true
+      },
+      success: @handle_data_response
+    });
+    @waiting_for_response = true;
+
+  handle_data_response: (response) =>
+    @more_labels_loadable = response.pagination.more;
+    @last_loaded_page += 1;
+    @waiting_for_response = false;
+
+    for new_label in response.results
+      @append_new_row(new_label.id, new_label.color, new_label.font_color, new_label.name, new_label.created_at, new_label.used_by_tasks);
+
+    @load_more_rows_if_necessary();
+
+  on_row_clicked: (event) =>
+    label_id = parseInt($(event.currentTarget).attr("label_id"));
+
+    if @selected_label_ids.includes(label_id)
+      @selected_label_ids = @selected_label_ids.filter (id) -> id != label_id
+      $(event.currentTarget).removeClass('selected');
+    else
+      @selected_label_ids.push(label_id);
+      $(event.currentTarget).addClass('selected');
+
+    if @selected_label_ids.length == 0
+      @merge_labels_input.val("");
+    else
+      most_used_selected_label_id = @selected_label_ids.reduce (max_id, id) => if @loaded_labels[id].used_by_tasks > @loaded_labels[max_id].used_by_tasks then id else max_id;
+      @merge_labels_input.val(@loaded_labels[most_used_selected_label_id].name);
+
+    @delete_labels_button.attr("disabled", @selected_label_ids.length == 0);
+    @merge_labels_button.attr("disabled", @selected_label_ids.length == 0);
+    @recolor_labels_button.attr("disabled", @selected_label_ids.length == 0);
+
+  sort_button_clicked: (sort_by) =>
+    $('.sort-by-id, .sort-by-name, .sort-by-created-at')
+      .removeClass('table-header-sort-down')
+      .removeClass('table-header-sort-up');
+
+    @sort_by_column = sort_by;
+
+    if @sort_by_order == 'desc'
+      @sort_by_order = 'asc';
+      $(event.target).addClass('table-header-sort-down');
+    else
+      @sort_by_order = 'desc';
+      $(event.target).addClass('table-header-sort-up');
+    @reload_table();
+
+  delete_selected_labels: () =>
+    confirm_msg = I18n.t('labels.index.delete_confirmation', {selected_labels_count: @selected_label_ids.length});
+
+    if (confirm(confirm_msg))
+      requests = []
+      for id in @selected_label_ids
+        requests.push($.ajax({url: "/labels/#{id}", type: "delete"}));
+      $.when.apply($, requests).then( () => @reload_table() );
+
+  merge_selected_labels: () =>
+    new_label_name = @merge_labels_input.val();
+    verification = verify_label_name(new_label_name);
+    if verification != 'OK'
+      alert(verification);
+      return;
+
+    selected_labels_count = @selected_label_ids.length;
+    confirm_msg = I18n.t('labels.index.merge_confirmation', {selected_labels_count: @selected_label_ids.length, new_merged_name: new_label_name})
+
+    if (confirm(confirm_msg))
+      $.ajax({
+        url: "/labels/merge",
+        type: "post",
+        data: {
+          label_ids: @selected_label_ids,
+          new_label_name: new_label_name
+        },
+        success: (response) => @reload_table();
```

Comment:
What about handling an error case in the frontend as well? An admin won't receive any information if updating doesn't work as expected.... (I think an alert would be enough).